temporian/core/operators/lag.py

```diff
@@ -84,21 +86,67 @@ def build_op_definition(cls) -> pb.OperatorDef:
 operator_lib.register_operator(LagOperator)
 
 
-def lag(event: Event, duration: Duration) -> Event:
-    if duration <= 0:
-        raise ValueError("duration must greater than zero")
+def _base_lag(
+    event: Event,
+    duration: Union[Duration, List[Duration]],
+    should_leak: bool = False,
+) -> Event:
+    """Base Lag function."""
 
-    return LagOperator(
-        event=event,
-        duration=duration,
-    ).outputs()["event"]
+    if isinstance(duration, (Duration, int)):
```

Comment:
(Duration, int) is written multiple times. Can you find a way to have a single definition of the supported types?